packages/block-library/src/legacy-widget/edit/widget-type-selector.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { Spinner, SelectControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+export default function WidgetTypeSelector( { selectedId, onSelect } ) {
+	const widgetTypes = useSelect( ( select ) => {
+		const hiddenIds =
+			select( blockEditorStore ).getSettings()
+				?.widgetTypesToHideFromLegacyWidgetBlock ?? [];
+		return select( coreStore )
+			.getWidgetTypes( { per_page: -1 } )
+			?.filter( ( widgetType ) => ! hiddenIds.includes( widgetType.id ) );
```

Comment:
I believe we also have to check if filter is a function too.
```suggestion
			?.filter?.( ( widgetType ) => ! hiddenIds.includes( widgetType.id ) );
```

Or maybe just create an intermediate variable?

Reply:
No this should be OK because if `getWidgetTypes()` returns something truthy then we know that it is an array and will contain `filter`.

Reply:
Oh no I totally misunderstood optional chaining then 😅 .